Switch to CUDA event based profiling

ipiszy opened the pull request:
In https://github.com/pytorch/pytorch/pull/107901, the CUDA event based
profiling is changed to profiler based profiling to avoid counting CPU-side
kernel launch overhead in final latency numbers. However, it turns out that
torch.profile() is significantly slower than CUDA event which affects model
compilation speed quite significantlly. This PR changes back to CUDA event
based profiling.

Follow-ups:
* Try CUDA event profiling with CUDAGraphs;
* Multi-GPU profiling;

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109338



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

ipiszy committed:
Switch to CUDA event based profiling

[ghstack-poisoned]

frank-wei reviewed (approved):
LGTM!

aakhundov reviewed (changes requested):
@ipiszy Should we also change these call sites: 

1. https://github.com/pytorch/pytorch/blob/0cbca85707eb95ad40babc4edadaf614a5ca86e1/torch/_inductor/select_algorithm.py#L648

2. https://github.com/pytorch/pytorch/blob/0cbca85707eb95ad40babc4edadaf614a5ca86e1/torch/_inductor/codegen/common.py#L1068

ipiszy commented:
@pytorchbot merge

ipiszy commented:
Thanks @adnanaziz ! Good catch! Sorry I missed these.

ipiszy committed:
Update on "Switch to CUDA event based profiling"

In https://github.com/pytorch/pytorch/pull/107901, the CUDA event based
profiling is changed to profiler based profiling to avoid counting CPU-side
kernel launch overhead in final latency numbers. However, it turns out that
torch.profile() is significantly slower than CUDA event which affects model
compilation speed quite significantlly. This PR changes back to CUDA event
based profiling.

Follow-ups:
* Try CUDA event profiling with CUDAGraphs;
* Multi-GPU profiling;




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

ipiszy commented:
@pytorchbot label "topic: not user facing"

ipiszy commented:
@pytorchbot merge

ipiszy committed:
Update on "Switch to CUDA event based profiling"

In https://github.com/pytorch/pytorch/pull/107901, the CUDA event based
profiling is changed to profiler based profiling to avoid counting CPU-side
kernel launch overhead in final latency numbers. However, it turns out that
torch.profile() is significantly slower than CUDA event which affects model
compilation speed quite significantlly. This PR changes back to CUDA event
based profiling.

Follow-ups:
* Try CUDA event profiling with CUDAGraphs;
* Multi-GPU profiling;




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

ipiszy commented:
@pytorchbot merge

ipiszy commented:
@pytorchbot merge

ipiszy commented:
@pytorchbot merge -f "unrelated failure"
